src/NGen.cpp

```diff
@@ -327,46 +329,109 @@ int main(int argc, char *argv[]) {
 
     std::shared_ptr<pdm03_struct> pdm_et_data = std::make_shared<pdm03_struct>(get_et_params());
 
-    //Now loop some time, iterate catchments, do stuff for total number of output times
-    for(int output_time_index = 0; output_time_index < manager->Simulation_Time_Object->get_total_output_times(); output_time_index++) {
-      //std::cout<<"Output Time Index: "<<output_time_index<<std::endl;
-      if(output_time_index%100 == 0) std::cout<<"Running timestep "<<output_time_index<<std::endl;
-      std::string current_timestamp = manager->Simulation_Time_Object->get_timestamp(output_time_index);
-      for(const auto& id : features.catchments()) {
-        //std::cout<<"Running cat "<<id<<std::endl;
-        auto r = features.catchment_at(id);
-        //TODO redesign to avoid this cast
-        auto r_c = dynamic_pointer_cast<realization::Catchment_Formulation>(r);
-        r_c->set_et_params(pdm_et_data);
-        double response = r_c->get_response(output_time_index, 3600.0);
-        std::string output = std::to_string(output_time_index)+","+current_timestamp+","+
-                             r_c->get_output_line_for_timestep(output_time_index)+"\n";
-        r_c->write_output(output);
-        //TODO put this somewhere else.  For now, just trying to ensure we get m^3/s into nexus output
-        try{
-          response *= (catchment_collection->get_feature(id)->get_property("areasqkm").as_real_number() * 1000000);
-        }catch(std::invalid_argument &e)
+    // check the time loops for the existing layers
+    ngen::LayerDataStorage& layer_meta_data = manager->get_layer_metadata();
+
+    // get the keys for the existing layers
+    std::vector<int>& keys = layer_meta_data.get_keys();
+
+    // get the converted time steps for layers
+    double min_time_step;
+    double max_time_step;
+    std::vector<double> time_steps;
+    for(int i = 0; i < keys.size(); ++i)
+    {
+        auto& m_data = layer_meta_data.get_layer(keys[i]);
+        double c_value = UnitsHelper::get_converted_value(m_data.time_step_units,m_data.time_step,"s");
+        time_steps.push_back(c_value);
+
+        if ( i == 0 )
         {
-          response *= (catchment_collection->get_feature(id)->get_property("area_sqkm").as_real_number() * 1000000);
+            min_time_step = max_time_step = c_value;
         }
-        //TODO put this somewhere else as well, for now, an implicit assumption is that a modules get_response returns
```

Comment:
Concerning that the web editor is flagging this section as a merge conflict when it was added 2 years ago?? Perhaps the changes are just so large that it's nice and confused.

Reply:
The merge conflict seems to have gone away--maybe my rebase fixed this.